mavisp/modules.py

```diff
@@ -105,63 +105,57 @@ def ingest(self, mutations):
-        for model in models:
-            method_dirs = os.listdir(os.path.join(self.data_dir, self.module_dir, f'{structure_ID}_{residue_range}', method, model))
-            if not set(method_dirs).issubset(set(self.methods.keys())):
-                this_error = f"One or more {self.name} methods are not supported"
-                raise MAVISpMultipleError(warning=warnings,
-                                        critical=[MAVISpCriticalError(this_error)])
-            all_methods.extend(method_dirs)
-
-        if len(all_methods) != len(set(all_methods)):
-                this_error = f"Only using one single instance of any given method is supported"
-                raise MAVISpMultipleError(warning=warnings,
-                                        critical=[MAVISpCriticalError(this_error)])
-
-        for model in models:
-
-            method_dirs = os.listdir(os.path.join(self.data_dir, self.module_dir, f'{structure_ID}_{residue_range}', method, model))
-
-            for method_dir in method_dirs:
-                self.methods[method_dir].parse(os.path.join(self.data_dir, self.module_dir, f'{structure_ID}_{residue_range}', method, model, method_dir))
-                self.data = self.data.join(self.methods[method_dir].data)
+            for model in models:
+                method_dirs = os.listdir(os.path.join(self.data_dir, self.module_dir, f'{structure_ID}_{residue_range}', method, model))
+                if not set(method_dirs).issubset(set(self.methods.keys())):
+                    this_error = f"One or more {self.name} methods are not supported"
+                    raise MAVISpMultipleError(warning=warnings,
+                                            critical=[MAVISpCriticalError(this_error)])
+                all_methods.extend(method_dirs)
+                if len(all_methods) != len(set(all_methods)):
+                                this_error = f"Only using one single instance of any given method is supported"
+                                raise MAVISpMultipleError(warning=warnings,
+                                                        critical=[MAVISpCriticalError(this_error)])
+            for model in models:
+
+                method_dirs = os.listdir(os.path.join(self.data_dir, self.module_dir, f'{structure_ID}_{residue_range}', method, model))
+                for method_dir in method_dirs:
+                    self.methods[method_dir].parse(os.path.join(self.data_dir, self.module_dir, f'{structure_ID}_{residue_range}', method, model, method_dir))
+                    self.data = self.data.join(self.methods[method_dir].data, rsuffix=" with  " + f'{method}' + " method")
 
         keys = [ k for k in self.data.keys() if k.startswith('Stability') ]
 
         # check if we have FoldX column
-        foldx_col = [k for k in keys if 'FoldX' in k]
-        if len(foldx_col) == 1:
-            foldx_header = foldx_col[0]
+        if any(['FoldX' in k for k in keys]):
```

Comment:
At the end of the calculation, we want to end up with **one classification per ensemble type (md, nmr...)**, not one classification in general. So I think all this part should be in a loop that runs over the ensemble type, in which for each ensemble type we 1) select the columns we need 2) do classification